pyiceberg/partitioning.py

```diff
@@ -413,7 +414,9 @@ def partition_record_value(partition_field: PartitionField, value: Any, schema:
     the final partition record value.
     """
     iceberg_type = schema.find_field(name_or_id=partition_field.source_id).field_type
-    iceberg_typed_value = _to_partition_representation(iceberg_type, value)
+    if not isinstance(value, int):
+        # When adding files, it can be that we still need to convert from logical types to physical types
+        iceberg_typed_value = _to_partition_representation(iceberg_type, value)
```

Comment:
is this due to the fact that we already transform the partition key value 
```
partition.transform.pyarrow_transform(source_field.field_type)(arrow_table[source_field.name])
```
and this expects the untransformed value?

if thats the case, can we just omit the transformation before the group_by?

Reply:
Ah, of course. We want to know the output tuples after the transform, so omitting the transformation is not possible. I think we could do a follow-up PR where we split out the logic for the write path, and the `add-files` path. Since after this PR, this is not needed when doing partitioned writes, we just need it to preprocess when importing partitions.